pkg/eventing/sources/v1alpha1/client.go

```diff
@@ -23,6 +27,10 @@ import (
 type KnSourcesClient interface {
 	// Namespace in which this client is operating for
 	Namespace() string
+
```

Comment:
For building sources, I wonder whether we should add an intermediate layer (like having an `KnApiServerSourceClient` which can be accessed from this interface). This would easily help in separation of concerns (as it this specialized client could be stored and picked up separately by the using code). This of course depends also on the amount of sources we expect, but even for 3 sources, I think this would be helpful (this is also how the generated service client does it).

I'm totally fine to keep it like this for the current PR but would probably refactor in a subsequent PR, when we e.. introduce the CronJobSource.

Reply:
Thanks.